src/main/java/org/sagebionetworks/bridge/hibernate/HibernateSurvey.java

```diff
@@ -0,0 +1,230 @@
+
+import java.util.List;
+
+import javax.persistence.CascadeType;
+import javax.persistence.CollectionTable;
+import javax.persistence.Column;
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinColumns;
+import javax.persistence.OneToMany;
+import javax.persistence.OrderColumn;
+import javax.persistence.Table;
+
+import org.hibernate.annotations.OnDelete;
+import org.hibernate.annotations.OnDeleteAction;
+import org.sagebionetworks.bridge.models.GuidCreatedOnVersionHolder;
+import org.sagebionetworks.bridge.models.surveys.Survey;
+import org.sagebionetworks.bridge.models.surveys.SurveyElement;
+import org.sagebionetworks.bridge.models.surveys.SurveyElementConstants;
+import org.sagebionetworks.bridge.models.surveys.SurveyId;
+import org.sagebionetworks.bridge.models.surveys.SurveyQuestion;
+import org.sagebionetworks.bridge.models.surveys.SurveySQL;
+
+import com.google.common.collect.ImmutableList;
+
+/** MySQL implementation of surveys via Hibernate. */
+@Entity
+@Table(name = "Surveys")
+@IdClass(SurveyId.class)
+public class HibernateSurvey implements Survey {
+    
+    private String studyKey;
+    @Id
+    private String guid;
+    @Id
+    private long createdOn;
+    private long modifiedOn;
+    private String copyrightNotice;
+    private String moduleId;
+    private Integer moduleVersion;
+    private Long version;
+    private String name;
+    private String identifier;
+    private boolean published;
+    private boolean deleted;
+    private Integer schemaRevision;
+    
+    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, 
+    fetch = FetchType.EAGER, targetEntity=HibernateSurveyElement.class)
+    @JoinColumns({
+       @JoinColumn(name = "surveyGuid", referencedColumnName = "guid"),
+       @JoinColumn(name = "createdOn", referencedColumnName = "createdOn")
+
+    })
+    private List<SurveyElement> elements;
+    
+    /**
+     * No args constructor, required and used by Hibernate for full object initialization.
+     */
+    public HibernateSurvey() {}
+    
+    public HibernateSurvey(Survey survey) {
+        this.studyKey = survey.getStudyIdentifier();
+        this.guid = survey.getGuid();
+        this.createdOn = survey.getCreatedOn();
+        this.modifiedOn = survey.getModifiedOn();
+        this.copyrightNotice = survey.getCopyrightNotice();
+        this.moduleId = survey.getModuleId();
+        this.moduleVersion = survey.getModuleVersion();
+        this.version = survey.getVersion();
+        this.name = survey.getName();
+        this.identifier = survey.getIdentifier();
+        this.published = survey.isPublished();
+        this.deleted = survey.isDeleted();
+        this.schemaRevision = survey.getSchemaRevision();
+        this.elements = survey.getElements();
```

Comment:
The Survey might be a DynamoSurvey, in which case the survey elements would be DynamoSurveyElements. We'll need a converter.

Also, what about converting SurveyElements to SurveyQuestions and SurveyInfoScreens?